include/con4m/gc.h

```diff
@@ -315,9 +311,40 @@ extern void           c4m_internal_lock_then_unstash_heap();
 extern void           c4m_get_heap_bounds(uint64_t *, uint64_t *, uint64_t *);
 extern void           c4m_gc_register_collect_fns(c4m_gc_hook, c4m_gc_hook);
 extern c4m_alloc_hdr *c4m_find_alloc(void *);
+extern bool           c4m_in_heap(void *);
 
 #ifdef C4M_GC_STATS
 uint64_t c4m_get_alloc_counter();
 #else
 #define c4m_get_alloc_counter() (0)
 #endif
+
+#ifdef C4M_FULL_MEMCHECK
+void c4m_alloc_display_front_guard_error(c4m_alloc_hdr *, void *, char *, int, bool);
+void c4m_alloc_display_rear_guard_error(c4m_alloc_hdr *, void *, int, void *, char *, int, bool);
+
+void _c4m_memcheck_raw_alloc(void *, char *, int);
+void _c4m_memcheck_object(c4m_obj_t, char *, int);
+#define c4m_memcheck_raw_alloc(x) \
+    _c4m_memcheck_raw_alloc(((void *)x), __FILE__, __LINE__);
+#define c4m_memcheck_object(x) \
+    _c4m_memcheck_object(((void *)x), __FILE__, __LINE__);
+#else
+#define c4m_memcheck_raw_alloc(x)
+#define c4m_memcheck_object(x)
+#endif
+
+#if __has_feature(address_sanitizer) || defined(__SANITIZE_ADDRESS__)
```

Comment:
gcc doesn't have `__has_feature`
This may be useful: https://stackoverflow.com/a/78444624

Reply:
Sucks that preprocessor doesn't short circuit. Fixed.